tasks/Functions-client/setAutoRequest.js

```diff
@@ -61,39 +90,49 @@ const setAutoRequest = async (contract, taskArgs) => {
 
   const requestConfig = getRequestConfig(unvalidatedRequestConfig)
 
-  console.log("\nEncrypting secrets and uploading to DON...")
-  const functionsRouterAddress = networks[network.name]["functionsRouter"]
-  const donId = networks[network.name]["donId"]
-  const signer = await ethers.getSigner()
-
-  const secretsManager = new SecretsManager({
-    signer,
-    functionsRouterAddress,
-    donId,
-  })
-
-  await secretsManager.initialize()
-  const encryptedSecretsObj = await secretsManager.buildEncryptedSecrets(requestConfig.secrets)
-  const slotId = parseInt(taskArgs.slotid)
-  const minutesUntilExpiration = 10 // Minimum 5 minutes supported.
-
-  const { version, success } = await secretsManager.uploadEncryptedSecretsToDON({
-    encryptedSecretsHexstring: encryptedSecretsObj.encryptedSecrets,
-    gatewayUrls: networks[network.name]["gatewayUrls"],
-    storageSlotId: slotId,
-    minutesUntilExpiration,
-  })
+  if (!requestConfig.secrets || Object.keys(requestConfig.secrets).length === 0) {
+    throw Error("\nThis task requires a secrets object in request config. None found")
```

Comment:
What if the user doesn't want to use secrets?  I'm not sure we should force them to use secrets here.

Reply:
In which case, slotId could be optional too.  We just need to verify it is present if the user has specified secrets.